Cargo.toml

```diff
@@ -1,26 +1,51 @@
-[package]
-name = "ttrpc"
-version = "0.8.4"
-authors = ["The AntFin Kata Team <kata@list.alibaba-inc.com>"]
+[workspace]
+members = [
+    "./compiler",
+    "./ttrpc-codegen",
+    "./example",
+    "./"
+]
+resolver = "2"
+
+[workspace.package]
 edition = "2018"
 license = "Apache-2.0"
-keywords = ["ttrpc", "protobuf", "rpc"]
 readme = "README.md"
 repository = "https://github.com/containerd/ttrpc-rust"
 homepage = "https://github.com/containerd/ttrpc-rust"
+authors = ["The AntFin Kata Team <kata@list.alibaba-inc.com>"]
+
+[workspace.dependencies]
```

Comment:
Why we need this section?

Reply:
The workspace allows you to share dependencies across multi crates (the workspace members).
It also makes things easier when crates in the project depend on each other, with e.g., the publish dry-run that takes into account other crates in the workspace.
For that to work, the shared dependencies need to be declared at the workspace level in this section.